src/ripple/conditions/Condition.h

```diff
@@ -77,36 +73,64 @@ class Condition
         This fingerprint is meant to be unique only with
         respect to other conditions of the same type.
     */
-    Buffer fingerprint;
+    std::array<std::uint8_t, 32> fingerprint;
 
     /** The cost associated with this condition. */
     std::uint32_t cost;
 
     /** For compound conditions, set of conditions includes */
-    std::set<Type> subtypes;
-
-    Condition(Type t, std::uint32_t c, Slice fp)
-        : type(t)
-        , fingerprint(fp)
-        , cost(c)
+    std::bitset<5> subtypes;
+
+    Condition(
+        Type t,
+        std::uint32_t c,
+        std::array<std::uint8_t, 32> const& fp,
+        std::bitset<5> const& s = std::bitset<5>{})
```

Comment:
Why not simply `std::bitset<5> s = {}`

Reply:
Fixed in b7a405a [fold] Address feedback from review: